src/main/java/com/cargohub/controllers/UserProfileController.java

```diff
@@ -0,0 +1,127 @@
+package com.cargohub.controllers;
+
+import com.cargohub.dto.BillingDetailsDto;
+import com.cargohub.dto.UserDto;
+import com.cargohub.entities.BillingDetailsEntity;
+import com.cargohub.entities.UserEntity;
+import com.cargohub.models.BillingDetailsModel;
+import com.cargohub.models.ResetPasswordModel;
+import com.cargohub.models.RestUserModel;
+import com.cargohub.models.UpdateUserModel;
+import com.cargohub.security.jwt.JwtTokenProvider;
+import com.cargohub.services.UserService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RestController
+@CrossOrigin
+@RequestMapping("/user/profile")
+//@Secured({"ROLE_ADMIN","ROLE_USER"})
+public class UserProfileController {
+
+    private UserService userService;
+    private ModelMapper modelMapper;
+
+    @Autowired
+    public UserProfileController(UserService userService, ModelMapper modelMapper) {
+        this.userService = userService;
+        this.modelMapper = modelMapper;
+
+    }
+
+
+    @CrossOrigin(origins = "http://localhost:3000")
+    @GetMapping(path = "/{id}",
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getUser(@PathVariable long id) {
+        try {
+            UserDto user = userService.getUserById(id);
+            if (user == null) {
+                throw new UsernameNotFoundException("User with id: " + id + " not found");
+            }
+            UpdateUserModel responseModel = modelMapper.map(user, UpdateUserModel.class);
+            return ResponseEntity.ok(responseModel);
+        } catch (AuthenticationException e) {
```

Comment:
it is not necessary to wrap all the secured controller methods logic in try/catch structure. Authentication are checking before servlets coming inside the controller method in filters of Spring Security. It is necessary in case of authentication process in the logic of controller method.